assets/app/scripts/controllers/build.js

```diff
@@ -69,6 +69,7 @@ angular.module('openshiftConsole')
                 };
               }
               $scope.build = build;
+              $scope.logCanRun = !(_.includes(['New', 'Pending'], build.status.phase));
```

Comment:
There won't be a log if the build phase is Error. The rest storage for logs will return an error: https://github.com/openshift/origin/blob/master/pkg%2Fbuild%2Fregistry%2Fbuildlog%2Frest.go#L116